tensorflow-ops/src/TensorFlow/Variable.hs

```diff
@@ -90,7 +90,11 @@ initializedVariable' :: forall a m v . (MonadBuild m, TensorType a)
                     => OpParams -> Tensor v a -> m (Variable a)
 initializedVariable' params initializer = do
     -- The shape is not known initially.
-    (Variable h Nothing :: Variable a) <- variableInternal params Nothing
+    variables <- variableInternal params Nothing
+    let (h, a) =
+          case variables of
+            (Variable h Nothing :: Variable a) -> (h, a)
+            _ -> undefined -- avoid MonadFail
```

Comment:
Could you make this a tad more helpful when it fails?
`error "variableInternal is empty"` should be easier to decipher.

Reply:
@blackgnezdo Thanks for the comments - good points all. Revised the patch, hopefully better.